ttf2mesh.c

```diff
@@ -676,12 +677,17 @@ int parse_simple_glyph(ttf_glyph_t *glyph, int glyph_index, uint8_t *p, int avai
 
     j = 0;
     n = -1;
+    int npoints_remaining = glyph->npoints;
     for (i = 0; i < glyph->ncontours; i++)
     {
         j = (int)big16toh(endPtsOfContours[i]);
         glyph->outline->cont[i].length = j - n;
         glyph->outline->cont[i].subglyph_id = glyph_index;
         glyph->outline->cont[i].subglyph_order = 0;
+
+        npoints_remaining -= glyph->outline->cont[i].length;
+        if (npoints_remaining < 0) return TTF_ERR_FMT;
```

Comment:
Not sure which causes the total amount of contour points to be overallocated - fixed by comparing contour-point count to total points.

Reply:
Yes you are right. TTF file with an invalid 'glyf' table may cause the parser to crash if there is no check.